Add nxcg Benchmark Algo Table Page (#75)
rapidsai/cugraph-docs: docs/cugraph-docs/source/nx_cugraph/benchmarks.rst

Benchmarks
=================

This page presents the performance of algorithms currently supported by ``nx-cugraph`` across four graphs of varying sizes. 

The goal is to provide a clear comparison of how dispatching to a GPU-accelerated backend compares against the default, CPU-based implementation of NetworkX. This allows users to get an idea of potential speedups that they might see by leveraging dispatching to a GPU with ``nx-cugraph``. 

> As datasets grow larger, the GPU-accelerated backend begins to show increasingly faster speedups over the CPU. This trend demonstrates how the GPU's parallel processing capabilities allow it to handle large-scale graph analytics much more efficiently than the CPU.

Reviewer: Maybe just put a qualifying word like "usually" or "typically", since different algos are sensitive to different graph characteristics and larger size isn't always a guarantee of more speedup.
```suggestion
As datasets grow larger, the GPU-accelerated backend typically begins to show increasingly faster speedups over the CPU. This trend demonstrates how the GPU's parallel processing capabilities allow it to handle large-scale graph analytics much more efficiently than the CPU.
```

      </tr>
      <tr>
        <td><a href="https://snap.stanford.edu/data/cit-Patents.html">cit-Patents</a></td>
        <td>3,774,768</td>
        <td>16,518,948</td>
        <td>A large, directed graph representing U.S. patent citation relationships.</td>
      </tr>
      <tr>
        <td><a href="https://snap.stanford.edu/data/soc-LiveJournal1.html">soc-LiveJournal1</a></td>
        <td>4,847,571</td>
        <td>68,993,773</td>
        <td>A very large, directed graph based on the LiveJournal social network.</td>
      </tr>
    </tbody>
  </table>
  </br>

--------------

Algorithm Table
***************

The table below showcases the performance difference between CPU and GPU when running a particular algorithm on one of the four datasets.

Each cell displays the speedup factor achieved when running the algorithm on a ``nx-cugraph`` compared to default NetworkX. A higher value indicates better performance. To enhance readability, the cells are color-coded based on the following speedup criteria:

- Green for excellent (``>5.0``)
- Yellow for moderate (``1.1 - 5.0``)
- Red for suboptimal (``0 - 1.0``)

> Also, by hovering over the table, users can see the time it took for NetworkX and nx-cugraph respectively, as well as the delta between both runs. We encourage users to refer to this table to see what kind of performance increase they can expect from their workflows.

Reviewer: Minor wording suggestion:
```suggestion
Also, by hovering over the table, users can see the time it took for NetworkX and nx-cugraph respectively, as well as the delta between both runs. We encourage users to refer to this table to see an approximation of the performance increase they can expect from their workflows.
```
